### ISSUE TITLE
bootstrap should retry if it fails to make consensus

### ISSUE THREAD

**Issue by david415** [katzenpost/authority#62]
if the bootstrap consensus fails then you have to wait a really long time for the next consensus. this is a bug.

**Comment by mixmasala** [comment]
tl;dr: bootstrap is for testing, not for rapidly deploying mix networks.

Because the authority protocol is dependent upon time there are some cases where bootstrapping won't work, e.g. after bootstrapping (for the current epoch) the next epoch voting round will already be past some deadline (descriptor submission, vote submission, etc) and can't be guaranteed to work - we just wait until the next epoch starts and follow the voting protocol as specified (voting for current epoch + 1). That means startup can take even longer, but behavior is well defined. Bootstrap behavior is not specified, and is intended to facilitate testing. The bootstrap process needs to converge to normal behavior and take into account a variety of situations such as:

* consensus already exists, enough authorities are online to make consensus
* there is no existing consensus, the network has just been started, but there's time to vote for the current epoch and the next epoch
* authorities are joining/leaving the consensus (flapping)
* authorities are started asynchronously

Successfully bootstrapping multiple authorities together requires starting them all at the same time, or otherwise synchronizing which state they start out in... this is normally relative-to-epoch but for bootstrap it happens to be relative to authority start.

In fact the bootstrap path complicates implementing the protocol as specified. My preference for deployment would be to remove the bootstrap path completely and use the time-based schedule to synchronize, because that is how the protocol is specified and KISS. If your deployment scenario requires rapidly starting mixnets, we need to revisit the authority specification.

**Comment by mixmasala** [comment]
We've removed the bootstrap path now and added support for setting the epochtime.Period at build or with go test.
